src/clj/harja/kyselyt/paallystys.sql

```diff
@@ -14,13 +14,15 @@ SELECT
   lahetetty,
   lahetys_onnistunut          AS "lahetys-onnistunut",
   lahetysvirhe
-FROM yllapitokohde
-  LEFT JOIN paallystysilmoitus pi ON pi.paallystyskohde = yllapitokohde.id
+FROM yllapitokohde ypk
+  LEFT JOIN paallystysilmoitus pi ON pi.paallystyskohde = ypk.id
                                      AND pi.poistettu IS NOT TRUE
 WHERE urakka = :urakka
       AND sopimus = :sopimus
       AND yllapitokohdetyotyyppi = 'paallystys' :: yllapitokohdetyotyyppi
-      AND yllapitokohde.poistettu IS NOT TRUE;
+      AND (:vuosi::INTEGER IS NULL OR (array_upper(vuodet, 1) IS NULL
+           OR vuodet @> ARRAY[:vuosi]::int[]))
```

Comment:
Onko tuo array_upper siis oikea tapa tarkistaa tuo? _"returns upper bound of the requested array dimension"_. En totta puhuen ihan ymmärrä, mitä tuo tarkoittaa.